lib/contracts/compiler.js

```diff
@@ -16,7 +16,9 @@ class Compiler {
 
     let pluginCompilers = self.plugins.getPluginsProperty('compilers', 'compilers');
     pluginCompilers.forEach(function (compilerObject) {
-      available_compilers[compilerObject.extension] = compilerObject.cb;
+      if(available_compilers[compilerObject.extension] === undefined){
```

Comment:
I don't understand, wouldn't this have the oposite behaviour described? i.e it will only override if it wasn't defined before, so if internally it defined .sol, then the plugin will not be able to override it

Reply:
Compiler plugins are loaded before the internal modules, so without this change, the internal module always overrides the plugin. Made some changes in latest commit to allow more than one compiler for the same extension and to have more clear code.